### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by dawudgovuk**
### Jira link (if applicable)
[DFR-2913](https://tools.hmcts.net/jira/browse/DFR-2913)

### Change description ###
FR Contested - INC5599921: Clear organisation policy issue

### Checklist
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java [hunk 1]
```diff
@@ -108,6 +108,39 @@ void givenACcdCallbackContestedCase_whenStartEventCalledAndAllPartiesAreDigitalA
...
+
+        List<DynamicMultiSelectListElement> dynamicElementList = List.of(getDynamicElementList(CaseRole.APP_SOLICITOR.getCcdCode()),
+            getDynamicElementList(CaseRole.RESP_SOLICITOR.getCcdCode()));
+
+        DynamicMultiSelectList parties = DynamicMultiSelectList.builder()
+            .value(dynamicElementList)
+            .listItems(dynamicElementList)
+            .build();
+
+        data.setPartiesOnCase(parties);
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java:131]: What's the purpose of setting this?

**Comment by jthmcts** [comment]
If a party has never been represented how does this work? How are they shown as an option?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java (hunk 1 above)
Comment (dawudgovuk) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java]: No purpose - was copied from above test. Removed now

**Comment by dawudgovuk** [comment]
> If a party has never been represented how does this work at the moment? How are they shown as an option?



> If a party has never been represented how does this work at the moment? How are they shown as an option?

When case is created, persistDefaultOrganisationPolicy is called to initialise with a default Organisation policy for unrepresented parties. Its only when Update Contact details event is called it clears this there is no Organisation Policy - so unrepresented parties are never shown as an option. I assume this is a bug as opposed to intended functionality.

**Comment by jthmcts** [comment]
> When case is created, persistDefaultOrganisationPolicy is called to initialise with a default Organisation policy for unrepresented parties. Its only when Update Contact details event is called it clears this there is no Organisation Policy - so unrepresented parties are never shown as an option. I assume this is a bug as opposed to intended functionality.

Could you look into the bug then rather than create a workaround that we might not need.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java [hunk 2]
```diff
@@ -108,6 +108,37 @@ void givenACcdCallbackContestedCase_whenStartEventCalledAndAllPartiesAreDigitalA
...
+        data.getContactDetailsWrapper().setApplicantFmName("Tony");
+        data.getContactDetailsWrapper().setApplicantLname("B");
+
+        data.getContactDetailsWrapper().setRespondentFmName("Tony");
+        data.getContactDetailsWrapper().setRespondentLname("C");
+
+        List<DynamicMultiSelectListElement> dynamicElementList = List.of(getDynamicElementList(CaseRole.APP_SOLICITOR.getCcdCode()),
+            getDynamicElementList(CaseRole.RESP_SOLICITOR.getCcdCode()));
+
+        DynamicMultiSelectList parties = DynamicMultiSelectList.builder()
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java:126]: This appears to be unused

Comment (dawudgovuk) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PartyServiceTest.java]: Removed this :)

**Review by jthmcts — APPROVED** [review]

**Pull request merged**